crates/router/src/types/storage/payment_intent.rs

```diff
@@ -121,4 +133,57 @@ impl PaymentIntentDbExt for PaymentIntent {
             .change_context(errors::DatabaseError::Others)
             .attach_printable("Error filtering records by time range")
     }
+
+    #[instrument(skip(conn))]
+    async fn apply_filters_on_payments(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+        constraints: &api::PaymentListFilterConstraints,
+    ) -> CustomResult<Vec<(Self, PaymentAttempt)>, errors::DatabaseError> {
+        let offset = constraints.offset.unwrap_or_default();
+        let mut filter = Self::table()
+            .inner_join(payment_attempt::table.on(dsl1::attempt_id.eq(dsl::active_attempt_id)))
+            .filter(dsl::merchant_id.eq(merchant_id.to_owned()))
+            .order(dsl::created_at.desc())
+            .into_boxed();
+
+        match &constraints.payment_id {
+            Some(pid) => {
+                filter = filter.filter(dsl::payment_id.eq(pid.to_owned()));
+            }
```

Comment:
```suggestion
            Some(payment_id) => {
                filter = filter.filter(dsl::payment_id.eq(payment_id.to_owned()));
            }
```

Prefer using verbose names.